SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKLogParser.cpp

```diff
@@ -0,0 +1,164 @@
+#include "Editor.h"
+#include "Internationalization/Regex.h"
+#include "Widgets/Input/SHyperlink.h"
+
+#define LOCTEXT_NAMESPACE "SpatialGDKLogParser"
+
+static void ShowMissingSchemaDialog(const TSet<FString>& MissingSchemaPaths)
+{
+	struct Local
+	{
+		static void OnHyperlinkClicked(const FString& InBlueprint, TSharedPtr<SCustomDialog> InDialog)
+		{
+			const FSoftObjectPath ObjectAtPath(InBlueprint);
+
+			UObject* ObjectToOpen = ObjectAtPath.ResolveObject();
+
+			if (!IsValid((ObjectToOpen)))
+			{
+				// The object could be unloaded; try and load it from the path.
+				ObjectToOpen = ObjectAtPath.TryLoad();
+			}
+
+			if (IsValid(ObjectToOpen))
+			{
+				if (ObjectToOpen->IsA<UBlueprintGeneratedClass>())
+				{
+					// BlueprintGeneratedClass can't be edited, but ClassGeneratedBy references
+					// the UBlueprint that has caused the class to be generated - open it instead.
+					ObjectToOpen = Cast<UBlueprintGeneratedClass>(ObjectToOpen)->ClassGeneratedBy;
+				}
+			}
+
+			if (IsValid(ObjectToOpen))
+			{
+				// Finally, if the object is available - ask the editor to edit it.
+				GEditor->EditObject(ObjectToOpen);
+			}
+
+			if (InDialog.IsValid())
+			{
+				// Opening the blueprint editor above may end up creating an invisible new window on top of the dialog,
+				// thus making it not interactable, so we have to force the dialog back to the front
+				InDialog->BringToFront(true);
+			}
+		}
+	};
+
+	TSharedRef<SVerticalBox> DialogContents =
+		SNew(SVerticalBox)
+		+ SVerticalBox::Slot().Padding(
+			0, 0, 0, 16)[SNew(STextBlock)
+							 .Text(LOCTEXT("MissingSchemaDialog",
+										   "The following objects have missing schema, check the logs for more information."))];
+
+	TSharedPtr<SCustomDialog> CustomDialog;
+
+	for (const FString& MissingSchemaPath : MissingSchemaPaths)
+	{
+		DialogContents->AddSlot().AutoHeight().HAlign(
+			HAlign_Left)[SNew(SHyperlink)
+							 .OnNavigate(FSimpleDelegate::CreateLambda([MissingSchemaPath, &CustomDialog]() {
+								 Local::OnHyperlinkClicked(MissingSchemaPath, CustomDialog);
+							 }))
+							 .Text(FText::FromString(MissingSchemaPath))
+							 .ToolTipText(LOCTEXT("MissingSchemaDialogLinkTT", "Click to open the object"))];
+	}
+
+	const FText DialogTitle = LOCTEXT("MissingSchemaDialogTitle", "Missing Schema");
+
+	const FText OKText = LOCTEXT("MissingSchemaDialogOk", "OK");
+
+	CustomDialog = SNew(SCustomDialog).Title(DialogTitle).DialogContent(DialogContents).Buttons({ SCustomDialog::FButton(OKText) });
+
+	CustomDialog->ShowModal();
+}
+
+class FMissingSchemaLogParser : public FOutputDevice
+{
+public:
+	~FMissingSchemaLogParser()
+	{
+		if (bShouldReportErrors && Paths.Num() > 0)
+		{
+			ShowMissingSchemaDialog(Paths);
+		}
+	}
+
+	virtual void Serialize(const TCHAR* LogMessage, ELogVerbosity::Type Verbosity, const FName& Category) override
+	{
+		if (Category == TEXT("LogSpatialClassInfoManager") && Verbosity <= ELogVerbosity::Warning)
```

Comment:
If you make the text a static const FName, this comparison could be instantaneous (FName are interned strings).